repository/release.go

```diff
@@ -128,14 +128,26 @@ func (r *ReleaseRepository) UpdateRelease(
 	return rls, nil
 }
 
+func (r *ReleaseRepository) DeleteReleaseByGitTag(ctx context.Context, githubOwnerSlug, githubRepoSlug, gitTag string) error {
+	return r.deleteRelease(ctx, query.DeleteReleaseByGitTag, pgx.NamedArgs{
+		"ownerSlug":  githubOwnerSlug,
+		"repoSlug":   githubRepoSlug,
+		"gitTagName": gitTag,
+	})
+}
+
 func (r *ReleaseRepository) DeleteRelease(ctx context.Context, projectID, releaseID uuid.UUID) error {
 	// Project ID is not needed in the query because releaseID is primary key
 	// But it is added for security reasons
 	// To make sure that the release belongs to the project that is passed from the service
-	result, err := r.dbpool.Exec(ctx, query.DeleteRelease, pgx.NamedArgs{
+	return r.deleteRelease(ctx, query.DeleteRelease, pgx.NamedArgs{
 		"projectID": projectID,
 		"releaseID": releaseID,
 	})
+}
+
+func (r *ReleaseRepository) deleteRelease(ctx context.Context, deleteQuery string, args pgx.NamedArgs) error {
```

Comment:
Absolutely understand the strive, but be careful with this.
Yes, you refactored it, with its benefits, etc., but simple metrics to be aware of:
- I spent like 2 minutes here (instead of 10 seconds in case it was repeated code).

Next time it might be less, but still, more than 10 seconds.

And then if you have a team of 10 people, over a year, it's like hours of difference in terms of time spent reading and reasoning about the code.

You can leave it like that, but might be an interesting thing to point out in the thesis.

Also, we are in Go. And Go was designed to scale, mainly in terms of number of engineers using it (you have a simple, unambiguous style guides etc.).

And again, we are in Go, and one idiomatic way to solve such things (from my understanding and experience, plus from what I saw in huge programs written in Go, for example [GCP Terraform provider by Google itself](https://github.com/hashicorp/terraform-provider-google?tab=readme-ov-file#developing-the-provider))..
One idiomatic way to solve this, is to generate the code.

But that's a whole different story of course.

And here it's obviously an overkill. 😄

Reply:
Thanks!